core/src/main/java/org/apache/accumulo/core/client/admin/CloneConfigurationImpl.java

```diff
@@ -0,0 +1,123 @@
+
+/**
+ * A {@link CloneConfiguration} implementation which also implements the builder thereof
+ *
+ * @since 1.10 and 2.1
+ */
+public class CloneConfigurationImpl implements CloneConfiguration, CloneConfiguration.Builder {
+
+  // The purpose of this is to allow building an immutable CloneConfiguration object without
+  // creating
+  // separate Builder and CloneConfiguration objects. This is done to reduce object creation and
+  // copying. This could easily be changed to two objects without changing the interfaces.
+  private boolean built = false;
+
+  // determines if memory is flushed in the source table before cloning.
+  private boolean flush = true;
+
+  // the sources table properties are copied, this allows overriding of those properties
+  private Map<String,String> propertiesToSet = null;
+
+  // do not copy these properties from the source table, just revert to system defaults
+  private Set<String> propertiesToExclude = null;
+
+  // do not bring the table online after cloning
+  private boolean keepOffline = false;
+
+  private CloneConfigurationImpl() {}
+
+  public boolean isFlush() {
+    Preconditions.checkState(built);
+    return flush;
+  }
+
+  public Map<String,String> getPropertiesToSet() {
+    Preconditions.checkState(built);
+    return (propertiesToSet == null ? Collections.<String,String>emptyMap()
+        : Collections.unmodifiableMap(propertiesToSet));
+  }
+
+  public Set<String> getPropertiesToExclude() {
+    Preconditions.checkState(built);
+    return (propertiesToExclude == null ? Collections.<String>emptySet()
+        : Collections.unmodifiableSet(propertiesToExclude));
+  }
+
+  public boolean isKeepOffline() {
+    Preconditions.checkState(built);
+    return keepOffline;
+  }
+
+  @Override
+  public Builder setFlush(boolean flush) {
+    Preconditions.checkState(!built);
+    this.flush = flush;
+    return this;
+  }
+
+  @Override
+  public Builder setPropertiesToSet(Map<String,String> propertiesToSet) {
+    Preconditions.checkState(!built);
+    this.propertiesToSet = propertiesToSet;
+    return this;
+  }
+
+  @Override
+  public Builder setPropertiesToExclude(Set<String> propertiesToExclude) {
+    Preconditions.checkState(!built);
+    this.propertiesToExclude = propertiesToExclude;
+    return this;
+  }
+
+  @Override
+  public Builder setKeepOffline(boolean keepOffline) {
+    Preconditions.checkState(!built);
+    this.keepOffline = keepOffline;
+    return this;
+  }
+
+  @Override
+  public CloneConfiguration build() {
+    Preconditions.checkState(!built);
+    built = true;
+    return this;
+  }
+
+  @Override
+  public String toString() {
+    return "{flush=" + flush + ", propertiesToSet=" + propertiesToSet + ", propertiesToExclude="
+        + propertiesToExclude + ", keepOffline=" + keepOffline + ", built=" + built + "}";
+  }
+
+  /**
+   * @return a {@link CloneConfiguration} builder
+   */
+  public static CloneConfiguration.Builder builder() {
```

Comment:
This method should be in the ```CloneConfiguration``` interface so users don't need to touch the Impl class.  With Java 1.8 you can make this a ```default``` method too.

Reply:
If the pom was not forcing 1.7 source compatibility, then I would.  In the top level pom of the 1.9 branch I see:

    <maven.compiler.release>7</maven.compiler.release>
    <maven.compiler.source>1.7</maven.compiler.source>
    <maven.compiler.target>1.7</maven.compiler.target>

Reply:
Probably need to update your branch... I added Java 1.8 in 9c957ab55f28f10dde1321eb613a9ec3e2fa5322 for the 1.10 release.

Reply:
I see it now.... ok moving static method into interface